biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/resolve/BndrunContainer.java

```diff
@@ -263,8 +265,22 @@ public void setRunrequiresFromProjectArtifact(Run run) {
 
 			Artifact artifact = project.getArtifact();
 
+			String bsn = artifact.getArtifactId();
+
+			try {
+				Domain domain = Domain.domain(artifact.getFile());
```

Comment:
domain can be null. Are you counting on the NPE dereferencing domain ending up in the catch block?

Reply:
how about:
```
			Artifact artifact = project.getArtifact();

			String bsn = artifact.getArtifactId();

			try {
				bsn = Optional.ofNullable(artifact.getFile())
					.map(asFunction(Domain::domain))
					.map(Domain::getBundleSymbolicName)
					.map(Map.Entry::getKey)
					.orElseGet(artifact::getArtifactId);
			} catch (Exception e) {
				logger.warn(
					"Could not get the Bundle-SymbolicName from the project artifact {}, falling back to artifactId {}",
					artifact, artifact.getArtifactId(), e);
			}
```

Reply:
That's fine, but you will not log any warning if the jar has no manifest or no BSN which I though you wanted to log.

Reply:
not precisely, if the project HAS the bnd-maven-plugin (already checked) and the artifact IS NOT a bundle, then something went wrong, no? Isn't that a different problem entirely? the resolver would fail to find the bundle since it couldn't be indexed as a proper OSGi bundle. It was not my intention to capture all those scenarios from this specific logic whose only job is to find a suitable value for inferred `-runrequires`.

Reply:
so the question is... should it be capturing all those cases?

Reply:
I've updated with new logic... let's go from that.

Reply:
I don't think so. I was just pointing out that your original commit logged for all those case. I think we only need to log for the exception case (as long as we don't NPE).